### PR TITLE
Run tests against a real server, not mock data

### PR DESCRIPTION
**Pull request by oyvindhagberg**
This commit does the following:
- Removes "playback" functionality from `mocktraffic.py` and renames the whole thing to a more appropriate name `recordhttp.py`
- Changes the format of recorded output slightly, now the whole file is valid json. Updates `testsuite-result.json` accordingly.
- Makes a few small changes to `testsuite`
- Adds a docker-compose-file to be used when running the tests
- Adds a shell script that runs the tests and compares api calls and output with a previously recorded file
- Moves everything related to CI into a new folder
- Modifies the workflow to use the aforementioned shell script

### CODE REVIEW

File: ci/run_testsuite_and_record.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# clean up on exit, even if something fails
+function cleanup {
+    docker compose down
+}
+trap cleanup EXIT
+
+# start mreg+postgres in containers
+docker compose up -d
+
+# create a superuser
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply ci/run_testsuite_and_record.sh:16]: That must be implemented in the [mreg repo](https://github.com/unioslo/mreg), we should create an issue there and do it later.

File: mreg_cli/recordhttp.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            return up.path
+
+    """ Records an http call (method, url and postdata) and the response. """
+    def record(self, method, url, params, data, result):
+        if not self.is_recording():
+            return
+        x = {
+            'method': method.upper(),
+            'url': self.urlpath(url, params),
+            'data': data,
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/recordhttp.py:96]: The test suite doesn't contain anything sensitive, and the recording function has no practical use outside of testing.

File: ci/run_testsuite_and_record.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (terjekv) [reply ci/run_testsuite_and_record.sh:16]: It's repos all the way down. I'll make an issue for mreg.

[the remark this answers is not in the document]
Comment (terjekv) [reply ci/run_testsuite_and_record.sh:16]: https://github.com/unioslo/mreg/issues/497

File: ci/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - MREG_DB_PASSWORD=mreg
+      - CI=yes
+    healthcheck:
+      test: sh -c "[ -f /var/run/gunicorn.pid ]"
+      interval: 5s
+      timeout: 5s
+      retries: 10
+      start_period: 5s
+
+  dummy-container:
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply ci/docker-compose.yml:38]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/b6277416deaeeaed42f5518b579aa4ab8882a12a

File: mreg_cli/recordhttp.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        RecordHttp.__instance.recorded_data.append(x)
+
+    def record_output(self,output):
+        if not self.is_recording():
+            return
+        x = {'output':output}
+        RecordHttp.__instance.recorded_data.append(x)
+
+    """ Returns only the path + query string components of a url """
+    def urlpath(self, url, params):
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/recordhttp.py]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/32dc12dbe5773795d76c70020a49c697caaae113

File: mreg_cli/recordhttp.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    def save_recording(self):
+        i = RecordHttp.__instance
+        f = open(i.filename, "w")
+        f.write(json.dumps(i.recorded_data, indent=2))
+        f.close()
+
+    """ Start recording http traffic, commands and console output to the given filename.
+        Warning! If the file exists, it will be deleted/overwritten. """
+    def start_recording(self, filename):
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/recordhttp.py]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/32dc12dbe5773795d76c70020a49c697caaae113

File: mreg_cli/main.py [hunk 6]
```diff
@@ -107,33 +107,24 @@ def main():
...
         log.logfile = conf['logfile']
 
-    m = mocktraffic.MockTraffic()
-    if 'mock_traffic' in conf:
-        if 'record_traffic' in conf:
-            print("You can't use both the playback and record options at the same time!")
-            raise SystemExit()
-        m.start_playback(conf['mock_traffic'])
-    elif 'record_traffic' in conf:
+    m = recordhttp.RecordHttp()
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/main.py]: Fixed in https://github.com/unioslo/mreg-cli/commit/2cba45ead6daa82ec1400ba3eb28bc2aa1a64e14

File: mreg_cli/recordhttp.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        cmd = cmd.lstrip()
+        if cmd.find("#")>-1:
+            cmd = cmd[0:cmd.find("#")].rstrip()
+        # don't log empty commands
+        if cmd == '':
+            return
+        x = {'command':cmd}
+        RecordHttp.__instance.recorded_data.append(x)
+
+    def record_output(self,output):
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/recordhttp.py]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/32dc12dbe5773795d76c70020a49c697caaae113

File: mreg_cli/recordhttp.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        if params:
+            url = f"{url}?{urlencode(params)}"
+        up = urlparse(url)
+        if up.query != '':
+            return up.path + '?' + up.query
+        else:
+            return up.path
+
+    """ Records an http call (method, url and postdata) and the response. """
+    def record(self, method, url, params, data, result):
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/recordhttp.py]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/aedb112324b74ac2c0e3df6ca4e9dfb67e9e8b77

File: mreg_cli/main.py [hunk 9]
```diff
@@ -160,14 +151,10 @@ def main():
...
-    if m.is_playback():
-        while True:
-            line = mocktraffic.MockTraffic().get_next_command()
-            if not line:
-                raise SystemExit()
-            print(">",line)
-            cli.parse(shlex.split(line))
+    # if the --source parameter was given, read commands from the source file and then exit
+    if 'source' in conf:
+        source([conf['source']],'verbosity' in conf,False)
```
[the remark this answers is not in the document]
Comment (oyvindhagberg) [reply mreg_cli/main.py]: Fixed in https://github.com/unioslo/mreg-cli/pull/165/commits/aedb112324b74ac2c0e3df6ca4e9dfb67e9e8b77

**Review by terjekv — APPROVED** [review]
Looks good to me!

**Pull request merged**